Source/missiles.cpp

```diff
@@ -1009,23 +950,14 @@ bool MonsterTrapHit(int m, int mindam, int maxdam, int dist, missile_id t, bool
 		monster._mhitpoints = 0;
 #endif
 	if (monster._mhitpoints >> 6 <= 0) {
-		if (monster._mmode == MonsterMode::Petrified) {
-			M_StartKill(m, -1);
-			monster.Petrify();
-		} else {
-			M_StartKill(m, -1);
-		}
+		M_StartKill(m, -1);
+	} else if (resist) {
+		PlayEffect(monster, 1);
 	} else {
-		if (resist) {
-			PlayEffect(monster, 1);
-		} else if (monster._mmode == MonsterMode::Petrified) {
-			if (m > MAX_PLRS - 1)
-				M_StartHit(m, -1, dam);
+		if (m > MAX_PLRS - 1)
+			M_StartHit(m, -1, dam);
+		if (monster._mmode == MonsterMode::Petrified)
 			monster.Petrify();

```

Comment:
This again appears to have had no effect, and the proposal defiantly won't, so can just be removed.